lib/puppet/http/client.rb

```diff
@@ -77,40 +85,63 @@ def head(url, headers: {}, params: {}, user: nil, password: nil, ssl_context: ni
 
     request = Net::HTTP::Head.new(url, @default_headers.merge(headers))
 
+    user = options.fetch(:user, nil)
+    password = options.fetch(:password, nil)
+    ssl_context = options.fetch(:ssl_context, nil)
+    include_system_store = options.fetch(:include_system_store, false)
+
     execute_streaming(request, user: user, password: password, ssl_context: ssl_context, include_system_store: include_system_store) do |response|
       response.body
     end
   end
 
-  def put(url, headers: {}, params: {}, content_type:, body:, user: nil, password: nil, ssl_context: nil, include_system_store: false)
+  def put(url, headers: {}, params: {}, options: {})
     query = encode_params(params)
     unless query.empty?
       url = url.dup
       url.query = query
     end
 
     request = Net::HTTP::Put.new(url, @default_headers.merge(headers))
+
+    body = options.fetch(:body)
+    content_type = options.fetch(:content_type)
+
```

Comment:
Previously ruby validated that all required keyword arguments were passed, and generated the following exception if arguments were omitted to `put` or `post`:

```
irb(main):004:0> client = Puppet::HTTP::Client.new
...
irb(main):005:0> client.put(URI("https://foo.com"))
ArgumentError (missing keywords: content_type, body)
```

When switching to `Hash#fetch` ruby raises a different `KeyError`, but the error message seems less helpful to me:

```
irb(main):004:0> client.put(URI("http://foo.com"))
Traceback (most recent call last):
        4: from /usr/local/opt/rbenv/versions/2.5.7/bin/irb:11:in `<main>'
        3: from (irb):4
        2: from /Users/josh/work/puppet/lib/puppet/http/client.rb:107:in `put'
        1: from /Users/josh/work/puppet/lib/puppet/http/client.rb:107:in `fetch'
KeyError (key not found: :body)
```

I'm thinking it would be better to raise ArgumentError explicitly for this case, either passing a block to `fetch` or calling `Hash#[]` and checking for a nil value? 

```
irb(main):006:0> {}.fetch(:body) { |_| raise ArgumentError, "put requires a 'body' option" }
Traceback (most recent call last):
        4: from /usr/local/opt/rbenv/versions/2.5.7/bin/irb:11:in `<main>'
        3: from (irb):6
        2: from (irb):6:in `fetch'
        1: from (irb):6:in `block in irb_binding'
ArgumentError (put requires a 'body' option)
```

Also could you add tests that verify we raise when required options are omitted? I think only `put` and `post` fall into that category?